java/server/src/org/openqa/grid/internal/RemoteProxy.java

```diff
@@ -39,6 +39,16 @@
  * only support Firefox.
  */
 public interface RemoteProxy extends Comparable<RemoteProxy> {
+
+  /**
+   *
+   * @param protocol - A {@link SeleniumProtocol} object that identifies the request flavor.
+   * @param capabilities - the type of test the client is interested in performing.
+   * @return - The entity on a proxy that can host a test session.
+   */
+  default TestSlot newTestSlot(SeleniumProtocol protocol, Map<String, Object> capabilities) {
```

Comment:
@mach6 

> I'd like to see a test that exercises the changes to TestSlot, overriding the default behaviour.

What exactly would we be asserting in these tests ? A downstream user is ideally speaking going to be sub-classing `TestSlot` and then work with that sub-class by accessing it via the `TestSession`. Can you please elaborate the unit test requirements here ?